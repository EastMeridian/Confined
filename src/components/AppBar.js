import React from 'react';
import t from 'prop-types';
import sizing from 'styles/sizing';

import View from './View';

const containerStyle = {
  flexDirection: 'row',
  justifyContent: 'space-between',
  alignItems: 'center',
  height: sizing.appBarHeight,
  backgroundColor: 'white',
  paddingLeft: '2em',
  paddingRight: '2em',
};

const AppBar = ({ children }) => (
  <View style={containerStyle}>
    {children}
  </View>
);

AppBar.propTypes = {
  children: t.node,
};

AppBar.defaultProps = {
  children: null,
};

export default AppBar;
