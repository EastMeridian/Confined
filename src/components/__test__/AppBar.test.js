import '@testing-library/jest-dom';
import { create } from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import AppBar from '../AppBar';

describe('AppBar component', () => {
  it('should matches the snapshot', () => {
    const appBar = create(<AppBar />);
    expect(appBar.toJSON()).toMatchSnapshot();
  });

  it('should display children', () => {
    const children = 'test';
    const { getByText } = render(<AppBar>{children}</AppBar>);
    expect(getByText(children)).toBeInTheDocument();
  });
});
