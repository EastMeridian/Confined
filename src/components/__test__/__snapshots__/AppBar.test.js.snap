// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar component should matches the snapshot 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "display": "flex",
      "flexDirection": "row",
      "height": "4em",
      "justifyContent": "space-between",
      "paddingLeft": "2em",
      "paddingRight": "2em",
    }
  }
/>
`;
